### PR TITLE
Enable music shuffling.

### PR DESCRIPTION
**Pull request by madprogrammi**
This is a fix for #14379. It no longer ignores the "shuffle" value in musicset.json.

### CODE REVIEW

File: src/sdltiles.cpp [hunk 1]
```diff
@@ -2041,7 +2041,18 @@ void musicFinished() {
     }
 
     // Load the next file to play.
-    current_playlist_at++;
+    if( list.shuffle ) {
+        size_t next_playlist_at = current_playlist_at;
+        // We'll retry until we get a random number that isn't our current one.
+        while( next_playlist_at == current_playlist_at ) {
```
Comment (BevapDin) [inline src/sdltiles.cpp]: Going to be fun with a playlist that contains only one entry (fun as in "infinite loop").

Btw. in my opinion, if you add a restriction, it's not random anymore.

**Comment by madprogrammi** [comment]
@BevapDin Good point. I was undecided if it should be allowed to repeat songs. I didn't catch the infinite loop though.

**Comment by kevingranade** [comment]
If you want it to have the behavior you're looking for, what you generally
want to do is make a secondary list of indices, shuffle those, then consume
them as you play the songs.

**Comment by sparr** [comment]
@madprogrammi instead of "fix for #" use "fixes #" next time so the issue is automatically closed.
https://help.github.com/articles/closing-issues-via-commit-messages/

**Pull request merged**